Sources/Utilities/DemoManager/DemoManager.swift

```diff
@@ -0,0 +1,154 @@
+
+extension DemoManager: DemoManaging {
+
+    public var isRunning: Bool { isEnabled }
+
+    public func start() {
+        isEnabled = true
+    }
+
+    public func stop() {
+        isEnabled = false
+    }
+
+    public func reset() {
+        DependencyContainer.Registrations.reset()
+        // Remove stubs if you're using them
+    }
+}
+
+// MARK: - Helpers
+
+private extension DemoManager {
+
+    func setupDemoEnvironment() {
+        mockDependencies()
+        // Also setup stubs if you're using them
+    }
+
+    func mockDependencies() {
+        // Mock any dependecy you need
+        DependencyContainer.complexSigletonClass.register(factory: { MockComplexSingletonClass() })
+    }
+}
+
+// MARK: - Stubbing -
+
+private extension DemoManager {
+
+    func stubResponse(
+        containing urlPart: String,
+        queryParamPart: String? = nil,
+        requiredHeaderValues: String?...,
+        bodyPart: [String]? = nil,
+        method: HTTPMethod? = nil,
+        statusCode: Int32 = 200,
+        from fileName: String? = nil
+        ) {
+        stub(
+            condition: { (request: URLRequest) -> Bool in
+                // Method
+                if let method = method?.rawValue.uppercased(), request.httpMethod?.uppercased() != method { return false }
+
+                // URL, query
+                if !(request.url?.absoluteString.contains(urlPart) ?? false) { return false }
+                if let part = queryParamPart, !(request.url?.absoluteString.contains(part) ?? false) { return false }
+
+                // Body
+                if
+                    let parts = bodyPart,
+                    let data = request.ohhttpStubs_httpBody,
+                    let body = String(data: data, encoding: .utf8)
+                {
+                    let bodyPartsPresent = parts
+                        .map { body.contains($0) }
+                        .contains(true)
+                    if !bodyPartsPresent { return false }
+                }
+
+                // Headers
+                let headers = request.allHTTPHeaderFields ?? [:]
+                let headersOk = requiredHeaderValues
+                    .compactMap { $0 }
+                    .map { headerValue in
+                        return headers.contains { $0.value == headerValue }
+                    }
+                    .allSatisfy { $0 }
+
+                return headersOk
+            },
+            response: { _ -> HTTPStubsResponse in
+                guard let fileName = fileName else {
+                    return HTTPStubsResponse(data: Data(), statusCode: statusCode, headers: nil)
+                }
+
+                let path = Bundle.main.path(forResource: fileName, ofType: "json") ?? ""
+
+                let headers = fileName == "locations" ?
+                    ["Content-Type": "text/html;charset=UTF-8"] :
+                    ["Content-Type": "application/json"]
+
+                if !Thread.isMainThread { sleep(1) }
```

Comment:
I don't have nothing against this although I would prefer if this function was asynchronous and has additional parameter `responseDelay: DispathTimeInternval` so we don't have to use sleep or similar. specially if delay depends on the thread.